File: backend/BookStore/BookStore.Domain/Entities/Order.cs
using BookStore.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookStore.Domain.Entities
{
    public class Order
    {
        public Guid Id { get; set; }
        public required string UserId { get; set; }
        public ApplicationUser? User { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        public ICollection<OrderItem> Items { get; set; } = [];
    }
}

File: frontend/bookstore-crm/src/app/core/models/order.ts
import { OrderStatus } from "./enums/order-status";
import { OrderItem } from "./order-item";
import { User } from "./user"

export interface Order {
    id: string
    user: User
    createdAt: Date;
    updatedAt: Date;
    status: OrderStatus;
    items: OrderItem[];
    totalPrice: number;
}